modules/slf4j/src/test/java/org/apache/ignite/logger/slf4j/Slf4jLoggerSelfTest.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2020 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.logger.slf4j;
+
+import java.io.File;
+import java.util.Collections;
+import java.util.Objects;
+import java.util.logging.Logger;
+import java.util.stream.Stream;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.ignite.Ignite;
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.apache.ignite.internal.util.typedef.G;
+import org.apache.ignite.internal.util.typedef.internal.U;
+import org.apache.ignite.spi.discovery.tcp.TcpDiscoverySpi;
+import org.apache.ignite.spi.discovery.tcp.ipfinder.vm.TcpDiscoveryVmIpFinder;
+import org.apache.logging.log4j.LogManager;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.slf4j.LoggerFactory;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Grid Slf4j SPI test.
+ */
+public class Slf4jLoggerSelfTest {
+    /** Path to full log. */
+    private static final String LOG_ALL = "work/log/all.log";
+
+    /** Path to filtered log. */
+    private static final String LOG_FILTERED = "work/log/filtered.log";
+
+    /** */
+    @Before
+    public void setUp() {
+        deleteLogs();
+
+        LogManager.shutdown();
+    }
+
+    /** */
+    @After
+    public void tearDown() {
+        deleteLogs();
+    }
+
+    /**
+     * Check that JUL is redirected to Slf4j .
+     *
+     * Start the local node and check presence of log file.
+     * Check that this is really a log of a started node.
+     * Log something using JUL logging.
+     * Check that logs present in log file.
+     * Log something in INFO level from package that is blocked in log4j2 configuration using JUL logging.
```

Comment:
Why do we say about log4j2 here? Is SLF backed by log4j? Let's clarify it somewhere in the comments.

Reply:
Fixed to slf4j. I don't think such details should be in the test description.